### PR TITLE
Document Box::into_raw returns non-null ptr

### LINKED ISSUE #52806: Box::into_raw doesn't document that the returned pointer will not be null
The existence of the unstable `Box::into_raw_non_null` kind of hints that this is the case (as does an understanding of how Box internally works), but it would be nice for unsafe code to actually guarantee this.

### PR DESCRIPTION
**Pull request by estk**
Closes  #52806.

r? @sfackler

### CODE REVIEW

File: src/liballoc/boxed.rs [hunk 1]
```diff
@@ -126,7 +126,8 @@ impl<T: ?Sized> Box<T> {
         Box(Unique::new_unchecked(raw))
     }
 
-    /// Consumes the `Box`, returning the wrapped raw pointer.
+    /// Consumes the `Box`, returning the wrapped raw pointer, which
+    /// will be properly aligned and non-null.
```
Comment (steveklabnik) [inline src/liballoc/boxed.rs]: So, `rustdoc` uses the first line as a summary, and this ends up being a bit long. Could you maybe change it to

```
/// Consumes the `Box<T>`, returning a raw pointer.
///
/// The pointer will be properly aligned and non-null.
```

**Review by steveklabnik — CHANGES_REQUESTED** [review]
Just a small formatting nit, thank you so much!

File: src/liballoc/boxed.rs (hunk 1 above)
Comment (estk) [reply src/liballoc/boxed.rs]: Thanks @steveklabnik, I didn't know that.

**Review by steveklabnik — APPROVED** [review]
Thank you!

**Pull request merged**